skyplane/api/api.py

```diff
@@ -0,0 +1,134 @@
+from dataclasses import dataclass, field
+from threading import Thread
+from typing import Dict, List, Optional, Set, Tuple
+
+from skyplane.api.auth_config import AWSConfig, AzureConfig, GCPConfig
+from skyplane.api.dataplane import Dataplane
+from skyplane.api.impl.planner import DirectPlanner
+from skyplane.api.impl.transfer_job import TransferJob, CopyJob, SyncJob
+from skyplane.api.impl.provisioner import Provisioner
+from skyplane.chunk import ChunkRequest
+from skyplane.obj_store.object_store_interface import ObjectStoreObject
+from skyplane.utils import logger
+
+# types
+TransferList = List[Tuple[ObjectStoreObject, ObjectStoreObject]]
+
+
+@dataclass
+class TransferProgressTracker(Thread):
+    dataplane: Dataplane
+    jobs: List[TransferJob]
+
+    # config
+    multipart_enabled: bool = True
+    multipart_threshold_mb: int = 128
+    multipart_chunk_size_mb: int = 64
+    multipart_max_chunks: int = 10000
+
+    # chunk_requests state
+    job_chunk_requests: Dict[TransferJob, List[ChunkRequest]] = field(default_factory=dict)
+    job_pending_chunk_ids: Dict[TransferJob, Set[int]] = field(default_factory=dict)
+    job_complete_chunk_ids: Dict[TransferJob, Set[int]] = field(default_factory=dict)
+
+    def run(self):
+        job_chunk_request_gen = {}
+        for job in self.jobs:
+            job_chunk_request_gen[job] = list(
+                job.dispatch(
```

Comment:
Will different jobs conflict due to shared chunk IDs? Should we migrate from linear IDs to UUIDs for chunks?

Reply:
Track via https://github.com/skyplane-project/skyplane/issues/643